pkg/kubectl/cmd/expose.go

```diff
@@ -253,12 +254,20 @@ func RunExpose(f cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []stri
 			}
 		}
 		info.Refresh(object, true)
-		if cmdutil.GetDryRunFlag(cmd) {
-			if len(cmdutil.GetFlagString(cmd, "output")) > 0 {
-				return f.PrintObject(cmd, false, mapper, object, out)
+		outputFormat := cmdutil.GetFlagString(cmd, "output")
+		dryRunFlag := cmdutil.GetDryRunFlag(cmd)
+		if cmdutil.GetFlagBool(cmd, "show-labels") {
```

Comment:
These errors should be checked before we fetch data from the server, not after.

Reply:
Always check flag preconditions prior to taking user action.

Reply:
Modified.
And, would you PTAL: :smile: 
> kindly request approve for: 
[Add unit test](https://github.com/kubernetes/kubernetes/pull/55930) for PR55371(fix show-labels output).
[Add validation for run option #56387](https://github.com/kubernetes/kubernetes/pull/56387), I add the `show-labels` check there for `kubectl run` too.